miden/src/tools/mod.rs

```diff
@@ -149,19 +208,34 @@ pub fn analyze<A>(
     program: &str,
     stack_inputs: StackInputs,
     advice_provider: A,
+    cycles_info: bool,
 ) -> Result<ExecutionDetails, ProgramError>
 where
-    A: AdviceProvider,
+    A: AdviceProvider + Clone,
 {
     let program = Assembler::default()
         .with_debug_mode(true)
         .with_library(&StdLibrary::default())
         .map_err(ProgramError::AssemblyError)?
         .compile(program)
         .map_err(ProgramError::AssemblyError)?;
-    let vm_state_iterator = processor::execute_iter(&program, stack_inputs, advice_provider);
+
     let mut execution_details = ExecutionDetails::default();
 
+    // if the `cycles_info` flag is true, get the program execution trace to obtain the trace length summary
+    if cycles_info {
+        let trace = processor::execute(
+            &program,
+            stack_inputs.clone(),
+            advice_provider.clone(),
+            ExecutionOptions::default(),
+        )
+        .map_err(|err| ProgramError::ExecutionError(err))?;
+        execution_details.set_extended_cycle_info(trace.trace_len_summary());
+    }
+
+    let vm_state_iterator = processor::execute_iter(&program, stack_inputs, advice_provider);
```

Comment:
Doing `processor:execute()` will cause us to execute the program twice. Is there a way to modify `vm_state_iterator` to get that info from it?

Reply:
Hmm, first I wanted to say that we can obtain info about operation only from `execute_iter` and info about VM cycles only from trace, which we can get from `execute`, but now I think we can get around it. I can try to implement another function, something like `execute_with_iter`, from which I can obtain both iterator and execution trace. Sorry @bobbinth, my bad, I'll rework this

Reply:
You can probably just modify the [VmStateIterator](https://github.com/0xPolygonMiden/miden-vm/blob/next/processor/src/debug.rs#L50) to expose this extra property.